File: database/QueryRecipesByTitle.js
var AWS = require("aws-sdk");

AWS.config.update({
  region: "us-east-1",
  endpoint: "https://dynamodb.us-east-1.amazonaws.com"
});

var docClient = new AWS.DynamoDB.DocumentClient();
var title = "cheesecake";

console.log("Querying for recipes with title " + title);

var params = {
    TableName : "Recipes",
    KeyConditionExpression: "#tle = :tttt",
    ExpressionAttributeNames:{
        "#tle": "title"
    },
    ExpressionAttributeValues: {
        ":tttt":title
    }
};

docClient.query(params, function(err, data) {
    if (err) {
        console.error("Unable to query. Error:", JSON.stringify(err, null, 2));
    } else {
        console.log("Query succeeded.");
        data.Items.forEach(function(item) {
            console.log(" -", item.year + ": " + item.title);
        });
    }
});

File: database/QueryRecipesByTitle.py
import boto3
import json
import decimal
from boto3.dynamodb.conditions import Key, Attr

# Helper class to convert a DynamoDB item to JSON.
class DecimalEncoder(json.JSONEncoder):
    def default(self, o):
        if isinstance(o, decimal.Decimal):
            if o % 1 > 0:
                return float(o)
            else:
                return int(o)
        return super(DecimalEncoder, self).default(o)

dynamodb = boto3.resource('dynamodb', region_name='us-east-1', endpoint_url="http://localhost:8000")
table = dynamodb.Table('Recipes')

query = "pasta"

response = table.query(
    KeyConditionExpression=Key('title').eq(query)
)

print 'Query returned %d results.' % len(response['Items'])
for i in response['Items']:
    print 'Recipe: %s, Creator: %s' % (i['title'], i['creator'])
